src/response_decoder.ts

```diff
@@ -84,6 +99,11 @@ function decodeResponse(
     }
     return { size, key, version, metadataInfo } as RawMetadataUpdateResponse
   }
+  if (key === CreditResponse.key) {
+    const responseCodeCredit = dataResponse.readUInt16()
+    const subscriptionId = dataResponse.readUInt8()
+    return { size, key, version, responseCode: responseCodeCredit, subscriptionId: subscriptionId } as RawCreditResponse
```

Comment:
I suggest a different management like:
```
    const response: RawDeliverResponse = {
      size,
      key: key as DeliverResponse["key"],
      version,
      subscriptionId,
      messages,
    }
return response
```
if you want to know  why @ggllcu will explain it.